src/workerd/jsg/setup.c++

```diff
@@ -360,15 +360,24 @@ IsolateBase::IsolateBase(const V8System& system, v8::Isolate::CreateParams&& cre
 
     ptr->GetHeapProfiler()->AddBuildEmbedderGraphCallback(buildEmbedderGraph, this);
 
-    // Create opaqueTemplate
     {
       // We don't need a v8::Locker here since there's no way another thread could be using the
       // isolate yet, but we do need v8::Isolate::Scope.
       v8::Isolate::Scope isolateScope(ptr);
       v8::HandleScope scope(ptr);
+
+      // Create opaqueTemplate
       auto opaqueTemplate = v8::FunctionTemplate::New(ptr, &throwIllegalConstructor);
       opaqueTemplate->InstanceTemplate()->SetInternalFieldCount(Wrappable::INTERNAL_FIELD_COUNT);
       this->opaqueTemplate.Reset(ptr, opaqueTemplate);
+
+      // Create Symbol.dispose and Symbol.asyncDispose.
+      symbolDispose.Reset(ptr, v8::Symbol::New(ptr,
+          v8::String::NewFromUtf8(ptr, "dispose",
+              v8::NewStringType::kInternalized).ToLocalChecked()));
+      symbolAsyncDispose.Reset(ptr, v8::Symbol::New(ptr,
+          v8::String::NewFromUtf8(ptr, "asyncDispose",
+              v8::NewStringType::kInternalized).ToLocalChecked()));
```

Comment:
`v8::Symbol::For(...)` is probably better here. And we should probably namespace the description in the polyfill, essentially like `Symbol.for('cloudflare.dispose')`.

Thinking about it, I guess there's a non-zero chance this might break someone if they happen to already be polyfilling `Symbol.dispose` in their worker. It's probably unlikely enough to not worry about, but it's something we'll want to keep an eye out for.

Reply:
> v8::Symbol::For(...) is probably better here.

Hmm, why would we want that? Seems like that could only lead to foot-shooting: someone might depend on the fact that `Symbol.dispose == Symbol.for("cloudflare.dispose")`, and then this assumption will break as soon as V8 starts providing this symbol itself.

> Thinking about it, I guess there's a non-zero chance this might break someone if they happen to already be polyfilling Symbol.dispose in their worker. It's probably unlikely enough to not worry about, but it's something we'll want to keep an eye out for.

How could it break someone?

If they are polyfilling carefully (only assigning the symbol if it doesn't already exist) they should be fine.

If they are just overwriting the symbol indiscriminately, they won't be able to use RPC correctly, but existing code should continue to work fine.

Reply:
Sorry, I had meant `ForApi(...)`. My brain got stuck on `Symbol.for` when I was writing the comment. If we used `ForApi(...)` we wouldn't need to cache the symbol ref anywhere and could just use `ForApi(...)` when we wanted to grab it again.... So if I took another pass at my comment, it would have been...

```
v8::Symbol::ForApi(...) is probably better....`
```

and

```
v8::Symbol::ForApi(js.strIntern("cloudflare.dispose"));
```

But as it is currently is fine also.

> How could it break someone

As I said, it's probably unlikely enough not to worry about, but theoretically someone could have a branch the executes only if `Symbol.dispose` is not present, or they are already depending on `Symbol.dispose` being identity equal to their own polyfilled `Symbol.for('whatever')` variant. 

I don't think anyone will actually be broken by this, just acknowledging that the risk is non-zero (even if super close to zero)

Reply:
Ah, I suppose `ForApi()` wouldn't have the problem of allowing applications to become dependent on it -- but still not clear if it provides any real advantages. The cached symbol ref is faster to access anyway (no need to look up an interned string, etc.) and more type-safe.